runtime/vm/CRIUHelpers.cpp

```diff
@@ -128,6 +128,18 @@ isCheckpointAllowed(J9VMThread *currentThread)
 	return result;
 }
 
+BOOLEAN
+enableCRIUSecProvider(J9VMThread *currentThread)
+{
+	BOOLEAN result = TRUE;
+
+	if (isCRIUSupportEnabled(currentThread)) {
+		result = J9_ARE_NO_BITS_SET(currentThread->javaVM->checkpointState.flags, J9VM_CRIU_ENABLE_CRIU_SEC_PROVIDER);
+	}
```

Comment:
I don't think this is right; the result should be `TRUE` only if support is enabled *and* the flag is set:
```c
	BOOLEAN result = FALSE;

	if (isCRIUSupportEnabled(currentThread)) {
		result = J9_ARE_ANY_BITS_SET(currentThread->javaVM->checkpointState.flags, J9VM_CRIU_ENABLE_CRIU_SEC_PROVIDER);
	}
```

Reply:
Updated